Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -30,31 +47,293 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
                 preferredSize: .avatarThumbnailSize,
                 components: components
             )
+            presenceAvatarView.isOnlineIndicatorVisible = false
             return
         }
 
-        let (avatarURL, isOnlineIndicatorVisible): (URL?, Bool) = {
-            // Try to get the explicit channel avatar first
-            if let avatarURL = channel.imageURL {
-                return (avatarURL, false)
-            }
-
-            // TODO: https://stream-io.atlassian.net/browse/CIS-652
-            // this is just placeholder implementation:
-
-            let firstOtherMember = channel.lastActiveMembers
-                .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
-                .first(where: { $0.id != content.currentUserId })
-
-            return (firstOtherMember?.imageURL, firstOtherMember?.isOnline ?? false)
-        }()
-
+        loadAvatar(for: channel)
+    }
+    
+    open func loadAvatar(for channel: ChatChannel) {
+        // If the channel has an avatar set, load that avatar
+        if let channelAvatarUrl = channel.imageURL {
+            loadChannelAvatar(from: channelAvatarUrl)
+            return
+        }
+      
+        // Use the appropriate method to load avatar based on channel type
+        if channel.isDirectMessageChannel {
+            loadDirectMessageChannelAvatar(channel: channel)
+        } else {
+            loadMergedAvatars(channel: channel)
+        }
+    }
+    
+    /// Loads the avatar from the URL. This function is used when the channel has a non-nil `imageURL`
+    /// - Parameter url: The `imageURL` of the channel
+    open func loadChannelAvatar(from url: URL) {
         presenceAvatarView.avatarView.imageView.loadImage(
-            from: avatarURL,
+            from: url,
             placeholder: appearance.images.userAvatarPlaceholder4,
             preferredSize: .avatarThumbnailSize,
             components: components
         )
+    }
+    
+    /// Loads avatar for a directMessageChannel
+    /// - Parameter channel: The channel
+    open func loadDirectMessageChannelAvatar(channel: ChatChannel) {
+        let lastActiveMembers = channel.lastActiveMembers
```

Comment:
question to double-check: maximum amount of lastActiveMembers 
(if its up to 10 its fine, if its 100+ need improve this code)

sorting of large array can impact on performance, 
if needed please add the counting in background here

Reply:
What makes you think sorting an array with 100+ is going to cause performance issues? Especially in a language performant like swift? Maybe 100k+ yes, not 100 😅

Reply:
We displaying avatars in table view, i.e. each cell want to perform this operation, this can cause UI freezes on older devices, while we should always to be focused on reaching 60-120 FPS of rendering. 
Should be tested, but to be secured we can just move this to background.

Reply:
According to current configuration in`ChatClientConfig.lastActiveMembersLimit`, the value is just `5`. This can of course be changed by the client, but I do not think so it will go above 100, hence I ignored the sorting cost

Reply:
Even for each cell, this will perform in nanoseconds, don't think it will ever block the main thread ever